tests/e2e/performance/test_small_file_workload.py

```diff
@@ -105,152 +235,77 @@ def add_key(self, key, value):
-                        val = float('{:.2f}'.format(hit['_source'][key]))
-                        if self.managed_keys[key]['name'] in res.keys():
-                            res[self.managed_keys[key]['name']].append(val)
-                        else:
-                            res[self.managed_keys[key]['name']] = [val]
-                    except Exception:
-                        pass
-        res = self.aggregate_threads_results(res)
-        return res
-
-    def aggregate_threads_results(self, res):
+        try:
+            self.es.index(index=self.new_index,
+                          doc_type='_doc',
+                          body=self.results,
+                          id=self.uuid)
+        except esexp.ConnectionError:
+            log.error('can not write data to ES server')
+
+    def aggregate_host_results(self, host_res):
         """
-        Aggregation of one section of results, this can be threads in host,
-        hosts in sample, samples in test
+        Aggregation results from all hosts in single sample
+        (each host is actually a pod)
 
         Args:
-            res (dict) : dictionary of results
+            host_res (dict): dictionary of host results - user for multi hosts
+                             test
 
         Returns:
-            dict : dictionary with the aggregate results.
+            dict: dictionary of all the host results aggregated with the host
+                  results that provide as argument to one set of results
 
         """
-
         results = {}
-        for key in self.managed_keys.keys():
-            if self.managed_keys[key]['name'] in res.keys():
-                results[key] = self.managed_keys[key]['op'](
-                    res[self.managed_keys[key]['name']]
-                )
-
-        # This is the place to check in host (treads) deviation.
-
-        return results
-
-    def combine_results(self, results, clear):
-        """
-        Combine 2 or more results (hosts in sample / samples in test)
-        to one result.
+        log.info('Aggregating hosts results to one test-results')
+        if self.results['full-res'] == {}:
+            self.results['full-res'] = host_res
+            log.info('This is the first pod - return results as is')
+        else:
 
-        Args:
-            results (dict): dictionary of results to combine
-            clear (bool): return only combined results or not.
-                          True - return only combined results
-                          False - add the combine results to originals results
+            """
+            Each test can do some operations, so i am running loop on all
+            operations that we have in this particular test.
 
-        Returns:
-            dict : dictionary of results records
+            """
+            for op in self.results['operations']:
 
-        """
+                """
+                For each operation, I am looking for all interesting lines
```

Comment:
I think better to add small snipet of items we are parsing here.

Reply:
added example of the log file.